File: BuscaBinaria/Python/buscaBinaria.py

def buscaBinaria(lista, target):
    inicio = 0
    end = len(lista) -1

    #euqnauto o inicio não ultrapassar o fim.
    while inicio <= end:
        #Encontrar meio da lista
        mid = (inicio + end) // 2
        
        #Se o alvo for encontrado, retorna o indicio
        if lista[mid] == target:
            return mid

        #Se o alvo for menor que o elemento meio, ignora a metade superior
        elif lista[mid] > target:
            end = mid -1

        #Se o alvo for maior que o elemento meio, ignora a metade inferior
        else:
            inicio = mid + 1
    
    return None

File: BuscaBinaria/C/buscaBinaria.c
#include <stdio.h>

int buscaBinaria(int lista[], int tamanho, int alvo) {
    int inicio = 0;
    int fim = tamanho - 1;

    while (inicio <= fim) {
        int mid = (inicio + fim) / 2;

        if (lista[mid] == alvo) {
            return mid;
        } else if (lista[mid] > alvo) {
            fim = mid - 1;
        } else {
            inicio = mid + 1;
        }
    }

    return -1;
}